### PR TITLE
Add support for custom protected branches

### LINKED ISSUE #99: Allow deleting merge branches except some branches
We have an interesting use case where we merge feature branches to `master`, and then `master` to `production`. In this case, enabling `MERGE_DELETE_BRANCH` means it correctly deletes feature branches after merging them, but also deletes `master` after merging it.

I'd love it if we could add a feature where we could set `MERGE_DELETE_BRANCH` to `true`, but also have a `DELETE_BRANCHES` parameter, perhaps like:

```yml
MERGE_DELETE_BRANCH: true
DELETE_BRANCHES: *,!master
```

i.e., "delete all branches after merging, but not `master`"

### PR DESCRIPTION
**Pull request by AnandChowdhary**
This PR fixes #99. It adds an environment variable, `MERGE_DELETE_PROTECT_BRANCHES`, where users can specify an additional list of protected branches that are not deleted after merges, like so:

```yaml
jobs:
  automerge:
    runs-on: ubuntu-latest
    steps:
      - name: automerge
        uses: "pascalgn/automerge-action@master"
        env:
          MERGE_LABELS: "automerge,!work in progress"
          MERGE_DELETE_PROTECT_BRANCHES: "master,production"
```

### CODE REVIEW

**Comment by pascalgn** [comment]
Without having looked at the changes yet, congratulations for getting #100! 🎉🎉🎉

File: README.md [hunk 1]
```diff
@@ -182,17 +186,17 @@ Also, the following general options are supported:
...
+  GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+  MERGE_LABELS: "automerge,!work in progress"
+  MERGE_REMOVE_LABELS: "automerge"
+  MERGE_METHOD: "squash"
+  MERGE_COMMIT_MESSAGE: "pull-request-description"
+  MERGE_FORKS: "false"
+  MERGE_RETRIES: "6"
+  MERGE_RETRY_SLEEP: "10000"
+  UPDATE_LABELS: ""
+  UPDATE_METHOD: "rebase"
```
Comment (pascalgn) [inline README.md:199]: Happens to me all the time, but please leave the formatting, as it's easier for people to copy+paste that way :)

File: README.md [hunk 2]
```diff
@@ -133,6 +133,10 @@ The following merge options are supported:
   repositories. Set this option to `true` to automatically delete branches
   after they have been merged.
 
+- `MERGE_DELETE_PROTECT_BRANCHES`: A comma-separated list of branches that will not
```
Comment (pascalgn) [inline README.md:136]: MERGE_DELETE_PROTECT_BRANCHES to me sounds a bit like "also delete protected branches" or so, which doesn't make sense. How about MERGE_DELETE_BRANCH_FILTER or so?

File: lib/common.js [hunk 3]
```diff
@@ -97,6 +97,10 @@ function createConfig(env = {}) {
   const mergeRetries = parsePositiveInt("MERGE_RETRIES", 6);
   const mergeRetrySleep = parsePositiveInt("MERGE_RETRY_SLEEP", 5000);
   const mergeDeleteBranch = env.MERGE_DELETE_BRANCH === "true";
+  const protectBranches = parseMergeRemoveLabels(
```
Comment (pascalgn) [inline lib/common.js:100]: Variable should start with "merge" to indicate it's about merging (not updating). Also, parseMergeRemoveLabels should be renamed then, as it's not really labels which are parsed here

File: README.md (hunk 1 above)
Comment (AnandChowdhary) [reply README.md]: Oops, reverted!

File: lib/common.js (hunk 3 above)
Comment (AnandChowdhary) [reply lib/common.js]: Makes sense. I've renamed `protectBranches` to `mergeProtectBranches` and `parseMergeRemoveLabels` to `parseCommaSeparatedList`, which is clearer.

File: README.md (hunk 2 above)
Comment (AnandChowdhary) [reply README.md]: Changed to `MERGE_DELETE_BRANCH_FILTER`.

**Comment by AnandChowdhary** [comment]
Pinging @pascalgn to review the requested changes 😄

Another related question: Is there any way to tell this action to not merge a PR until all status checks have passed? Again, I cannot rely upon branch protection rules and am looking for a way to configure this action, maybe with `MERGE_PENDING_STATUS_CHECKS: false` and `MERGE_FAILED_STATUS_CHECKS: false`?

File: lib/common.js [hunk 4]
```diff
@@ -97,6 +100,10 @@ function createConfig(env = {}) {
   const mergeRetries = parsePositiveInt("MERGE_RETRIES", 6);
   const mergeRetrySleep = parsePositiveInt("MERGE_RETRY_SLEEP", 5000);
   const mergeDeleteBranch = env.MERGE_DELETE_BRANCH === "true";
+  const mergeProtectBranches = parseCommaSeparatedList(
```
Comment (pascalgn) [inline lib/common.js:103]: This should be renamed to match the MERGE_DELETE_BRANCH_FILTER name 🙏  (see other variables in lines 90-100)

**Comment by pascalgn** [comment]
@AnandChowdhary This PR currently has some conflicts, could you please have a look? 🙏

**Comment by pascalgn** [comment]
Somehow the build hasn't run (but I might have configured it incorrectly). Could you maybe try pushing to the branch again, so that the build might trigger? 🙏

**Comment by AnandChowdhary** [comment]
Ok, I just pushed a random commit. :)

**Comment by pascalgn** [comment]
The build failed because 1 of the tests seems to fail. Could you run `yarn` locally, it should also run the tests?

**Comment by RaghunadhR** [comment]
Can someone help with Github Action yml script for 

1. Adding 2 approvals
2. Auto Merge

a Pull Request with a pre-requisite of label "AutoApproveMerge" on it.

**Comment by pascalgn** [comment]
Thanks for the PR, it's merged now! 👍

**Pull request merged**